appveyor.yml

```diff
@@ -1,27 +1,29 @@
 environment:
   matrix:
-  - TOXENV: "linting,docs,doctesting"
-    PYTEST_NO_COVERAGE: "1"
   - TOXENV: "py27"
-  - TOXENV: "py34"
-  - TOXENV: "py35"
-  - TOXENV: "py36"
   - TOXENV: "py37"
+    PYTEST_NO_COVERAGE: "1"
+  - TOXENV: "linting,docs,doctesting"
+  - TOXENV: "py36"
+  - TOXENV: "py35"
+  - TOXENV: "py34"
   - TOXENV: "pypy"
     PYTEST_NO_COVERAGE: "1"
-  - TOXENV: "py27-xdist"
-  - TOXENV: "py27-trial"
-  - TOXENV: "py27-numpy"
+  # Specialized factors for py27.
+  - TOXENV: "py27-trial,py27-numpy,py27-nobyte"
   - TOXENV: "py27-pluggymaster"
     PYTEST_NO_COVERAGE: "1"
-  - TOXENV: "py36-xdist"
-  - TOXENV: "py36-trial"
-  - TOXENV: "py36-numpy"
+  - TOXENV: "py27-xdist"
+  # Specialized factors for py36.
+  - TOXENV: "py36-trial,py36-numpy"
   - TOXENV: "py36-pluggymaster"
     PYTEST_NO_COVERAGE: "1"
-  - TOXENV: "py27-nobyte"
   - TOXENV: "py36-freeze"
     PYTEST_NO_COVERAGE: "1"
+  - TOXENV: "py36-xdist"
+
+matrix:
+  fast_finish: true
```

Comment:
I'm not sure this is a good idea. If more than one environment will fail, I would like to get the full picture, otherwise we might get into a situation where you fix a job, then it fails on the next one, then you fix it only to discover it fails yet again...

Reply:
ok.  Will remove the 2nd commit then.

Reply:
OTOH you will get the full picture in Travis typically, and AppVeyor is more problematic with regard to build queue length.

After all I would even run less jobs altogether on AppVeyor (e.g. only py27 and py36 or py37, not py34 etc).  Maybe that could be configured for PRs?  (and still run everything on merges to master)

Reply:
> OTOH you will get the full picture in Travis typically, and AppVeyor is more problematic with regard to build queue length.

Typically yes, you are right. Well let's leave it like this, and see how it goes. Please just add a comment explaining the rationale for future reference, I think:

```yaml
matrix:
  # stop build if a job fails: usually we get the full picture from
  # travis anyway, and AppVeyor is problematic regarding build queue length
  # we can revert this if this proves problematic
  fast_finish: true
```

What do you think?

> Maybe that could be configured for PRs? (and still run everything on merges to master)

I'm not a big fan on running different sets of jobs in PRs vs master: this only leads to situation of a green PR breaking `master` because the PR did not run all jobs.

Reply:
Sounds good, will add the comment.